qiskit_aer/backends/aerbackend.py

```diff
@@ -427,12 +427,10 @@ def _execute_circuits_job(self, circuits, noise_model, config, job_id="", format
         metadata_list = []
         for idx, circ in enumerate(circuits):
             if circ.metadata:
-                metadata = circ.metadata
-                metadata_list.append(metadata)
-                circ.metadata = {}
-                circ.metadata["metadata_index"] = idx
+                metadata_list.append(circ.metadata)
+                circ.metadata = {"metadata_index": idx}
             else:
-                metadata_list.append(None)
+                metadata_list.append({})
```

Comment:
Sure I'm happy to do that. With this change, the line in question factors right out of the if statement,
```python 
        for idx, circ in enumerate(circuits):
            metadata_list.append(circ.metadata)
            if circ.metadata:
                circ.metadata = {"metadata_index": idx}
```
so I'm wondering how this will be able to be simplified further?

Reply:
89e8ba9f6eb2e5e08c056118de372e288697fe1a

(feel free to suggest other wording, I chickened out and did not give actual advice for how to change it later)